controller/state.go

```diff
@@ -947,6 +945,25 @@ func useDiffCache(noCache bool, manifestInfos []*apiclient.ManifestResponse, sou
 	return true
 }
 
+// specEqualsCompareTo compares the application spec to the comparedTo status. It normalizes the destination to match
+// the comparedTo destination before comparing. It does not mutate the original spec or comparedTo.
+func specEqualsCompareTo(spec v1alpha1.ApplicationSpec, comparedTo v1alpha1.ComparedTo) bool {
```

Comment:
Could be ptr arguments, but I dont think performance will increase that much.
```suggestion
func specEqualsCompareTo(spec *v1alpha1.ApplicationSpec, comparedTo *v1alpha1.ComparedTo) bool {
```